### PR TITLE
[docs] move NEWS to the documentation

### PR DESCRIPTION
**Pull request by odow**
The Python ecosystem do this. I think it works quite well: http://xarray.pydata.org/en/latest/whats-new.html

It's doubly important for this release of MOI because there are such a large number of changes.

### CODE REVIEW

**Comment by odow** [comment]
I think this is useful: https://jump.dev/MathOptInterface.jl/previews/PR1565/release_notes/

**Comment by blegat** [comment]
Why not keeping `NEWS.md` where it is ? So that it can be accessed directly in addition to being in the docs ? An advantage of having it in the docs is that we can use Documenter features such as `@ref`. In that case, we can move it in `docs/src` but then we would need a clear link to if from the README

**Comment by odow** [comment]
> Why not keeping NEWS.md where it is ? So that it can be accessed directly in addition to being in the docs ?

You can still link directly: https://github.com/jump-dev/MathOptInterface.jl/blob/od/release_notess/docs/src/release_notes.md

How often to people actually check the `NEWS.md` file? Adding it to the docs makes it much more visible. We could add a placeholder `NEWS.md` that just says "check the docs"?

> having it in the docs is that we can use Documenter features such as @ref

I thought about this, but then when we remove things like `SingleVariable` you just break all the previous links. We could get better at writing detailed release notes with links to the PRs through.

File: docs/src/release_notes.md [hunk 1]
```diff
@@ -1,8 +1,7 @@
-MathOptInterface (MOI) release notes
-====================================
+# Release notes
 
-v0.10.0 (In development)
-------------------------
+v0.10.0 (unreleased)
+---------------------------
 
 **MOI is a significant breaking release. There are a large number of
```
Comment (matbesancon) [inline docs/src/release_notes.md:6]:
```suggestion
**MOI v0.10 is a significant breaking release. There are a large number of
```

**Pull request merged**